src/encoded/static/components/static-pages/components/TableOfContents.js

```diff
@@ -267,20 +271,24 @@ class TableEntryChildren extends React.Component {
                 }
 
                 var hAttributes = MarkdownHeading.getAttributes(h.props.children);
-                var hString = TableOfContents.textFromReactChildren(h.props.children);
-                var link;
+                var linkTitle = TableOfContents.textFromReactChildren(h.props.children);
+
+                // We must have this to be equal to the ID of the element we're navigating to.
+                // A custom ID might be set in Markdown 'attributes' which we prefer over the one passed to explicitly via props.
+                var link = (hAttributes && hAttributes.id) || h.props.id || null;
+
                 if (hAttributes && hAttributes.matchedString){
-                    hString = hString.replace(hAttributes.matchedString, '').trim();
-                    if (hAttributes.id){
-                        link = hAttributes.id;
-                    }
+                    linkTitle = linkTitle.replace(hAttributes.matchedString, '').trim();
                 }
-                if (!link) link = TableOfContents.slugify(hString);
+
+                /** @deprecated */
+                if (!link) link = TableOfContents.slugify(linkTitle); // Fallback -- attempt to not use -- may fail.
+
```

Comment:
This block of code above is the primary 'fix'. It uses `h.props.id` instead of `link = TableOfContents.slugify(linkTitle)`.

`hAttributes.id` is an exceptional case where someone could define `{#someID.text-400}` below their Markdown heading and then `someID` gets parsed out as hAttributes.id to override header ID manually in content.